pandas/tests/groupby/test_function.py

```diff
@@ -1234,3 +1234,26 @@ def test_groupby_sum_timedelta_with_nat():
     res = gb["b"].sum(min_count=2)
     expected = Series([td3, pd.NaT], dtype="m8[ns]", name="b", index=expected.index)
     tm.assert_series_equal(res, expected)
+
+
+def test_groupby_empty_dataset():
+    # GH#41575
+    df = DataFrame(columns=["A", "B", "C"], dtype=int)
```

Comment:
df here should be non-empty; that way we are testing the non-empty case `df` against the empty case `df.iloc[:0]` below.

Reply:
I think it would also be good to have different columns with different dtypes for testing include/exclude: int, float, object would be sufficient.